lib/mayaUsd/ufe/UsdAttributeHolder.cpp

```diff
@@ -555,12 +555,69 @@ PXR_NS::SdfValueTypeName UsdAttributeHolder::usdAttributeType() const
 Ufe::AttributeEnumString::EnumValues UsdAttributeHolder::getEnumValues() const
 {
     Ufe::AttributeEnumString::EnumValues retVal;
+    if (_usdAttr.IsValid()) {
+        for (auto const& option : getEnums()) {
+            retVal.push_back(option.first);
+        }
+    }
+
+    return retVal;
+}
+
+UsdAttributeHolder::EnumOptions UsdAttributeHolder::getEnums() const
+{
+    UsdAttributeHolder::EnumOptions retVal;
     if (_usdAttr.IsValid()) {
         VtTokenArray allowedTokens;
         if (_usdAttr.GetPrim().GetPrimDefinition().GetPropertyMetadata(
                 _usdAttr.GetName(), SdfFieldKeys->AllowedTokens, &allowedTokens)) {
             for (auto const& token : allowedTokens) {
-                retVal.push_back(token.GetString());
+                retVal.emplace_back(token.GetString(), "");
+            }
+        }
+        // We might have a propagated enum copied into the created the NodeGraph port, resulting
```

Comment:
(minor) Grammar. Guessing just remove the "the".

Reply:
Fixed.